op-e2e/system/gastoken/gastoken_test.go

```diff
@@ -571,3 +270,375 @@ func waitForTx(t *testing.T, tx *types.Transaction, err error, client *ethclient
+	require.NoError(t, err)
+	withdrawalTx, err := feeVault.Withdraw(l2Opts)
+	waitForTx(t, withdrawalTx, err, l2Client)
+
+	// Get the receipt and the amount withdrawn
+	receipt, err = l2Client.TransactionReceipt(context.Background(), withdrawalTx.Hash())
+	require.NoError(t, err)
+
+	inclusionHeight := receipt.BlockNumber.Uint64()
+	it, err := feeVault.FilterWithdrawal(&bind.FilterOpts{
+		Start: inclusionHeight,
+		End:   &inclusionHeight,
+	})
+	require.NoError(t, err)
+	require.True(t, it.Next())
+
+	withdrawnAmount := it.Event.Value
+
+	// Finalize the withdrawal
+	proveReceipt, finalizeReceipt, resolveClaimReceipt, resolveReceipt := helpers.ProveAndFinalizeWithdrawal(t, cfg, gto.sys, "verifier", cfg.Secrets.Alice, receipt)
+	require.Equal(t, types.ReceiptStatusSuccessful, proveReceipt.Status)
+	require.Equal(t, types.ReceiptStatusSuccessful, finalizeReceipt.Status)
+	if allocType.UsesProofs() {
+		require.Equal(t, types.ReceiptStatusSuccessful, resolveClaimReceipt.Status)
+		require.Equal(t, types.ReceiptStatusSuccessful, resolveReceipt.Status)
+	}
+
+	// Assert that the recipient's balance did increase
+	var recipientBalanceAfter *big.Int
+	if enabled {
+		recipientBalanceAfter, err = weth9.BalanceOf(&bind.CallOpts{}, recipient)
+	} else {
+		recipientBalanceAfter, err = l1Client.BalanceAt(context.Background(), recipient, nil)
+	}
+	require.NoError(t, err)
+
+	require.Equal(t, recipientBalanceAfter, new(big.Int).Add(recipientBalanceBefore, withdrawnAmount))
+}
+
+func checkL1TokenNameAndSymbol(t *testing.T, gto gasTokenTestOpts, enabled bool) {
+	l1Client := gto.sys.NodeClient("l1")
+	cfg := gto.cfg
+	enabledExpectations := gto.enabledExpectations
+	disabledExpectations := gto.disabledExpectations
+
+	systemConfig, err := bindings.NewSystemConfig(cfg.L1Deployments.SystemConfigProxy, l1Client)
+	require.NoError(t, err)
+
+	token, err := systemConfig.GasPayingToken(&bind.CallOpts{})
+	require.NoError(t, err)
+
+	name, err := systemConfig.GasPayingTokenName(&bind.CallOpts{})
+	require.NoError(t, err)
+
+	symbol, err := systemConfig.GasPayingTokenSymbol(&bind.CallOpts{})
+	require.NoError(t, err)
+
+	if enabled {
+		require.Equal(t, enabledExpectations.tokenAddress, token.Addr)
+		require.Equal(t, enabledExpectations.tokenDecimals, token.Decimals)
+		require.Equal(t, enabledExpectations.tokenName, name)
+		require.Equal(t, enabledExpectations.tokenSymbol, symbol)
+	} else {
+		require.Equal(t, disabledExpectations.tokenAddress, token.Addr)
+		require.Equal(t, disabledExpectations.tokenDecimals, token.Decimals)
+		require.Equal(t, disabledExpectations.tokenName, name)
+		require.Equal(t, disabledExpectations.tokenSymbol, symbol)
+	}
+}
+
+func checkL2TokenNameAndSymbol(t *testing.T, gto gasTokenTestOpts, enabled bool) {
+	l2Client := gto.sys.NodeClient("sequencer")
+	enabledExpectations := gto.enabledExpectations
+	disabledExpectations := gto.disabledExpectations
```

Comment:
Non-blocking suggestion: instead of passing `enabled bool`, we could just pass `expectations`. Then the caller is responsible for passing either `enabledExpectations` or `disabledExpectations`,  and we can reduce the duplication / conditional flow around L609.

Reply:
Some of these other helpers could get the same treatment, although perhaps not all of them...

Reply:
updated this for the ones that made sense